src/daemon/db.rs

```diff
@@ -0,0 +1,45 @@
+use sqlx::sqlite::SqliteConnectOptions;
+use sqlx::Pool;
+use sqlx::Row;
+use sqlx::SqlitePool;
+use std::str::FromStr;
+
+use super::context::{CtxField, DaemonContext};
+
+static DATABASE: CtxField<SqlitePool> = |ctx| {
+    let options = SqliteConnectOptions::from_str(ctx.init().db_path.to_str().unwrap())
+        .unwrap()
+        .create_if_missing(true);
+
+    smol::future::block_on(async move {
+        let pool = Pool::connect_with(options).await.unwrap();
+        let _ = sqlx::query(
```

Comment:
this should panic if failed too, so use unwrap rather than throwing away the error